libcomposefs/lcfs-mount.c

```diff
@@ -578,34 +578,40 @@ static errint_t lcfs_mount_erofs_ovl(struct lcfs_mount_state_s *state,
 	char imagemountbuf[] = "/tmp/.composefs.XXXXXX";
 	char *imagemount;
 	bool created_tmpdir = false;
-	char loopname[PATH_MAX];
+	char source[PATH_MAX];
 	int errsv;
 	errint_t err;
 	int loopfd;
 
 	image_flags = lcfs_u32_from_file(header->flags);
 
-	loopfd = setup_loopback(state->fd, state->image_path, loopname);
-	if (loopfd < 0)
-		return loopfd;
-
 	if (options->image_mountdir) {
 		imagemount = (char *)options->image_mountdir;
 	} else {
 		imagemount = mkdtemp(imagemountbuf);
 		if (imagemount == NULL) {
 			errsv = errno;
-			close(loopfd);
 			return -errsv;
 		}
 		created_tmpdir = true;
 	}
 
-	err = lcfs_mount_erofs(loopname, imagemount, image_flags, state);
-	close(loopfd);
+	sprintf(source, "/dev/fd/%d", state->fd);
```

Comment:
`snprintf` please, even if it's "impossible" to overflow.

We usually say `/proc/self/fd/%d` there.

I'd be curious about how this shows up in `/proc/mounts`.  It's a purely theoretical question, though, since we unmount it very quickly.

Reply:
I don't think `sprintf` is a problem. `setup_loopback` uses it as well:
https://github.com/containers/composefs/blob/1003b4f20c93a3475ca4fc028c3396870a58db63/libcomposefs/lcfs-mount.c#L246

It's impossible the length of `/dev/fd/` plus the length of `%d` exceeds `PATH_MAX`.

Reply:
I agree it's not a real problem, and the readlink thing would be pretty gross.

Reply:
The `sprintf` thing is a problem, though, and we should probably fix the other user as well.

Reply:
> The `sprintf` thing is a problem, though, and we should probably fix the other user as well.

Okay then